src/KeyDB.php

```diff
@@ -88,10 +88,11 @@ public function __construct()
      */
     public function addKey(string $keyId, Key $key): void
     {
-        if (! in_array($key->getType(), self::getSupportedKeyTypes(), true)) {
+        $keyType = $key->getType();
+        if (! in_array($keyType, self::getSupportedKeyTypes(), true)) {
             // @todo Convert this to a log line as per Python.
             // https://github.com/php-tuf/php-tuf/issues/160
-            throw new InvalidKeyException("Root metadata file contains an unsupported key type: \"{$keyMeta['keytype']}\"");
```

Comment:
$keyMeta is an undefined variable. The fact that this never caused a test failure is proof positive that this code has never been exercised in our test cases until now.